providers/microshift/microshift.sh

```diff
@@ -0,0 +1,139 @@
+#
+# This file is part of the Kepler project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2022 The Kepler Contributors
+#
+
+set -eu -o pipefail
+
+declare -r MICROSHIFT_DEFAULT_NETWORK="cluster"
+declare -r MICROSHIFT_REGISTRY_NAME=${MICROSHIFT_REGISTRY_NAME:-registry}
+declare -r MICROSHIFT_IMAGE=${MICROSHIFT_IMAGE:-quay.io/microshift/microshift-aio}
+declare -r MICROSHIFT_TAG=${MICROSHIFT_TAG:-latest}
+declare -r MICROSHIFT_CONTAINER_NAME=${MICROSHIFT_CONTAINER_NAME:-microshift}
+declare -r MICROSHIFT_KUBECONFIG_DIR="$PROJECT_ROOT/tmp/microshift"
+declare -r MICROSHIFT_KUBECONFIG_FILE="$MICROSHIFT_KUBECONFIG_DIR/kubeconfig"
```

Comment:
If we are exporting the kubeconfig under `MICROSHIFT_KUBECONFIG_DIR` then I suggest we have to set KUBECONFIG env variable to `$MICROSHIFT_KUBECONFIG_FILE ` as kubectl/oc client will fail to pick up kubeconfig file when run in CI.